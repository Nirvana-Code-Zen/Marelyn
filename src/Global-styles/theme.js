export default {
  colors: {
    primary: '#d4a413',
    primary_complement: '#efb810',
    primary_second_complement: '#f3c332',
    secondary: '#155db1',
    secondary_complement: '#2196f3',
    error: '#D50000',
    white: '#ffffff',
    hover_search: '#efb810'
  },
  fonts: {
    texts_titles: '"Didact Gothic", sans-serif'
  },
  fonts_sizes: {
    titles: '50px',
    subtitles: '40px',
    small_subtitles: '30px',
    mobile_font: '25px',
    texts: '19px',
    medium_text: '17px',
    button_text: '16px',
    small_texts: '15px',
    extra_small_texts: '14px'
  },
  width_sizing: {
    wTen: '0.625rem', /* 10px */
    wTwelve: '0.75rem', /* 12px */
    Fourteen: '0.875rem', /* 14px */
    wFourteen: '1rem', /* 16px */
    wTwenty: '1.25rem', /* 20px */
    wTwentyFour: '1.5rem', /* 24px */
    wTwentyEight: '1.75rem', /* 28//px */
    wThirtyTwo: '36rem', /* 32px */
    wSixty: '160px',
    wThirteenthTundred: '300px'
  },
  width_porcentages: {
    wOneSixth: '16%',
    wOneFifth: '20%',
    wQuarter: '25%',
    wOneThrid: '33.333333%',
    wTwoFifths: '40%',
    wMedium: '50%',
    wTreeFifths: '60%',
    wTwoThrid: '66.666667%',
    wFourFifths: '80%',
    wTreeQuarters: '75%',
    wFiveSixths: '83.333333%',
    wElevenTwelfth: '91.666667%',
    wComplet: '100%'
  },
  width_creen: {
    wScreen: '100vw',
    wMin: 'min-content',
    wMax: 'max-content',
    wFit: 'fit-content',
    minW0: '0px'
  }
}
