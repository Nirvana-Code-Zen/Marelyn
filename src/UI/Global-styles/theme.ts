interface Theme {
  colors: {
    primary: string;
    primary_complement: string;
    primary_second_complement: string;
    secondary: string;
    secondary_complement: string;
    error: string;
    white: string;
    hover_search: string;
    texts: string;
    texts_secondary: string;
    hover_tabs: string;
  };
  fonts: {
    texts_titles: string;
  };
  fonts_sizes: {
    titles: string;
    subtitles: string;
    medio_titles: string;
    small_subtitles: string;
    texts: string;
    medium_text: string;
    button_text: string;
    small_texts: string;
    extra_small_texts: string;
  };
  width_sizing: {
    [key: string]: string; 
  };
  width_porcentages: {
    [key: string]: string;
};
width_creen:{
  [key:string]: string;
}
}


export const theme: Theme = {
  colors: {
    primary: '#d4a413',
    primary_complement: '#efb810',
    primary_second_complement: '#f3c332',
    secondary: '#155db1',
    secondary_complement: '#2196f3',
    error: '#D50000',
    white: '#ffffff',
    hover_search: '#efb810',
    texts: '#999999',
    texts_secondary: '#0B0A59',
    hover_tabs: '#E3D99F'
  },
  fonts: {
    texts_titles: '"Didact Gothic", sans-serif'
  },
  fonts_sizes: {
    titles: '30.12rem', /* 50px */
    subtitles: '2.5rem', /* 40px */
    medio_titles: '1.87rem', /* 30px */
    small_subtitles: '1.25rem', /* 20px */
    texts: '1.18rem', /* 19px */
    medium_text: '1.06rem', /* 17px */
    button_text: '1rem', /* `16px */
    small_texts: '0.93rem', /* 15px */
    extra_small_texts: '0.87rem'/* 14px */
  },
  width_sizing: {
    wTen: '0.625rem', /* 10px */
    wTwelve: '0.75rem', /* 12px */
    Fourteen: '0.875rem', /* 14px */
    wFourteen: '1rem', /* 16px */
    wTwenty: '1.25rem', /* 20px */
    wTwentyFour: '1.5rem', /* 24px */
    wTwentyEight: '1.75rem', /* 28//px */
    wThirtyTwo: '36rem', /* 32px */
    wSixty: '160px',
    wThirteenthTundred: '300px'
  },
  width_porcentages: {
    wOneSixth: '16%',
    wOneFifth: '20%',
    wQuarter: '25%',
    wOneThrid: '33.333333%',
    wTwoFifths: '40%',
    wMedium: '50%',
    wTreeFifths: '60%',
    wTwoThrid: '66.666667%',
    wFourFifths: '80%',
    wTreeQuarters: '75%',
    wFiveSixths: '83.333333%',
    wElevenTwelfth: '91.666667%',
    wComplet: '100%'
  },
  width_creen: {
    wScreen: '100vw',
    wMin: 'min-content',
    wMax: 'max-content',
    wFit: 'fit-content',
    minW0: '0px'
  }
}

