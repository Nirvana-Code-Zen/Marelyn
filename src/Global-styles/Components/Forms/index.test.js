import { ThemeProvider } from 'styled-components'
import Form, { Input, GroupForm } from './index'
import renderer from 'react-test-renderer'
import theme from '../../theme'

describe('<Form /> Styled Component', () => {
  test('Should be Form styles ready', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <Form />
      </ThemeProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Should be Input styles ready', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <Input />
      </ThemeProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Should be FormGroup styles ready', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <GroupForm />
      </ThemeProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
