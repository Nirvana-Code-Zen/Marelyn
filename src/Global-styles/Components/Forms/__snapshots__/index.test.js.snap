// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> Styled Component Should be Form styles ready 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: auto;
  margin-top: 0;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: w-one-sixth:16%;
  w-one-fifth: 20%;
  w-quarter: 25%;
  w-one-thrid: 33.333333%;
  w-two-fifths: 40%;
  w-medium: 50%;
  w-tree-fifths: 60%;
  w-two-thrid: 66.666667%;
  w-four-fifths: 80%;
  w-tree-quarters: 75%;
  w-five-sixths: 83.333333%;
  w-eleven-twelfth: 91.666667%;
  w-complet: 100%;
  row-gap: 0;
}

.c0 p {
  font-family: "Didact Gothic",sans-serif;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 15px;
  width: 75%;
  text-align: center;
  box-sizing: border-box;
  cursor: pointer;
}

.c0 span {
  font-family: "Didact Gothic",sans-serif;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 15px;
  text-align: center;
  box-sizing: border-box;
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: black;
  cursor: pointer;
}

<form
  className="c0"
/>
`;

exports[`<Form /> Styled Component Should be FormGroup styles ready 1`] = `
.c0 {
  position: relative;
  margin: 1rem 0;
}

.c0 .bar:before {
  left: 50%;
}

.c0 .bar:after {
  right: 50%;
}

.c0 .bar:before,
.c0 .bar:after {
  content: '';
  height: 2px;
  width: 0;
  bottom: 1px;
  position: absolute;
  background: #155db1;
  -webkit-transition: 0.2s ease all;
  transition: 0.2s ease all;
}

.c0 .bar:before:focus-within .label-char,
.c0 .bar:after:focus-within .label-char {
  -webkit-transform: translateY(-20px);
  -ms-transform: translateY(-20px);
  transform: translateY(-20px);
}

.c0 label {
  font-family: "Didact Gothic",sans-serif;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  color: #999;
  font-size: 19px;
  font-weight: normal;
  position: absolute;
  pointer-events: none;
  left: 5px;
  top: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0"
/>
`;

exports[`<Form /> Styled Component Should be Input styles ready 1`] = `
.c0 {
  font-size: 19px;
  padding: 10px 10px 10px 5px;
  display: block;
  width: 260px;
  border: none;
  border-bottom: 1px solid #d4a413;
  background: transparent;
  box-sizing: border-box;
}

.c0:focus {
  outline: none;
  border-bottom: none;
}

.c0:focus ~ label .label-char {
  -webkit-transform: translateY(-20px);
  -ms-transform: translateY(-20px);
  transform: translateY(-20px);
  font-size: 16px;
  color: #155db1;
}

.c0:focus ~ .bar:before,
.c0:focus ~ .bar:after {
  width: 50%;
}

<input
  className="c0"
/>
`;
